runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -301,13 +304,64 @@ public long commitTransaction() throws TransactionAbortedException {
         commitAddress = address;
 
         // Update all proxies, committing the new address.
-        updateAllProxies(x ->
-                x.getUnderlyingObject()
-                        .optimisticCommitUnsafe(commitAddress));
+        tryCommitAllProxies();
 
         return address;
     }
 
+
+    /** Try to commit the optimistic updates to each proxy. */
+    protected void tryCommitAllProxies() {
+        // First, get the committed entry
+        // in order to get the backpointers
+        // and the underlying SMREntries.
+        final ILogData committedEntry = this.builder.getRuntime()
+                .getAddressSpaceView().read(commitAddress);
+
+        if (committedEntry.getType() == DataType.EMPTY) {
+            return;
+        }
+
+        updateAllProxies(x -> {
+            // If some other client updated this object, sync
+            // it forward to grab those updates
+            if (getSnapshotTimestamp() !=
```

Comment:
individual streams backpointers might not be equal to the global snapshot. Did you mean 
     getSnapshotTS() < backpointer
?